DFC.Digital/DFC.Digital.Web.Sitefinity.Core/SitefinityExtensions/AssetLocationAndVersion.cs

```diff
@@ -66,11 +66,18 @@ private static string GetFileHash(string file)
 
         private async Task<string> GetFileHashAsync(string assetLocation)
         {
-            HttpResponseMessage response = await httpClientService.GetAsync(assetLocation);
-            if (response.IsSuccessStatusCode)
+            try
             {
-                var hashCode = response.Content.Headers.GetValues(ContentMDS).FirstOrDefault();
-                return hashCode.Replace("-", string.Empty);
+                var response = await httpClientService.GetAsync(assetLocation);
+                if (response.IsSuccessStatusCode)
+                {
+                    var hashCode = response.Content.Headers.GetValues(ContentMDS).FirstOrDefault();
+                    return !string.IsNullOrWhiteSpace(hashCode) ? hashCode.Replace("-", string.Empty) : DateTime.Now.ToString("yyyyMMddHH");
+                }
+            }
+            catch (Exception)
+            {
+                // ignored, it has been logged by exception interception
```

Comment:
We should still log the exception and then gracefully return datetime value

Reply:
Resolved